src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let copyOfSeparator = separator;
+
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === this[this.length - 1]) {
+        copyOfSeparator = '';
+      } else if (this[i] === null || this[i] === undefined) {
```

Comment:
<img width="216" alt="Screenshot 2020-08-06 at 14 32 48" src="https://user-images.githubusercontent.com/35385361/89527284-c034d700-d7f1-11ea-9627-428d0a9a0a66.png">

you need to check every element in an array, why you exclude the last one